### DISCUSSION TITLE
Community meeting Jan 12, 2021

### DISCUSSION THREAD

**Discussion by carlisia** [vmware-tanzu/velero#3301] [Community meetings]
# Jan 12, 2021
### Status Updates
-    dave
    -    e2e tests running in Azure
    -    Working on e2e tests running against all clouds as install target
- nrb
    - Out most of last week due to family emergency
    - Reviewed https://github.com/vmware-tanzu/velero/issues/3191 for the cause of throttling log messages
        - Increase --client-burst & --client-qps if you see this. We'll get this increased in v1.6
    - Opened an issue for some of Velero's upcoming rearchitecture, https://github.com/vmware-tanzu/velero/issues/3229
    - Covering community support w/ Carlisia this week
- bridget
    - Community support last week
    - Working on [multiple credentials support](https://github.com/vmware-tanzu/velero/pull/2403)
        - Need to make a decision on approach to take
    - Need to follow up on some PRs that have been updated
-    carlisia
    -    This week will be PR reviews and addressing code reviews on my PRs
    -    Sharing community support with Nolan
    -    PTO this Thu and Fri

### Discussion Topics
- dave
    - Azure issues with memory limit and plug-in failure.  Hard to debug some plug-in failures.
        - nrb - There is an issue w/ Azure plugin where object storage chunk uploading uses the maximum memory of the Velero pod.
        - nrb - Adding ~10 log messages into the Backup CR seems useful, especially when it can't upload to object store
- Dylan
    - Ability to selectively patch resources on per restore basis. 
        - Use case is Secrets & CRs
            - User wants to continually patch Secrets vs replace it
            - Operators watching CRs, they wanted to make sure the operators were going to keep working
        - Current design of plugins is very much all or nothing.
            - Plugins should be able to allow mutations or patching
        - Give post hooks the backup manifest to help
        - When patching, this is a merge operation. That may result in a merged, incorrect object at the end
        - Can we add a way to download individual JSON from a backup to then re-submit with Kubectl?
            - This exists with the `velero backup download` command, but you need to take it out of a tarball
        - Can we modify the restoreitemaction plugin API to return an action?
            - The plugin could identify the relevant fields and return a patch object & the patch action
        - Action - Alay and Dylan to work on design doc
- Phuong: (phuong.n.hoang@dell.com) invite to design meeting for Velero plugin timeout 9am PST Friday Jan 15th. This is the Zoom meeting: https://Dell.zoom.us/j/91078486282?pwd=Yk9sU1UwT0NScXE2Q2oyRGV3K002QT09 
    - This is for timing out plugins generically
    - Enhance the plugins to use the Context for timeouts
    - Please add to https://groups.google.com/g/projectvelero
- carlisia
    - hit a temporary snag with using GH Discussion: https://github.com/github/feedback/discussions/3105
    - is HackMD getting slow again? Time to archive?
- nrb
    - Current v1.6.0 status: https://github.com/vmware-tanzu/velero/projects/4
        - Testing is moving along
        - Need to get multiple credential support design approved
- bridget
    - Can we do a v1.5.3 release in the meantime before 1.6?
        - nrb: :thumbsup:
            - We need to include some patches like properly deleting when DeleteItemActions are missing, and fixing include/exclude plugin logic (I don't have a full list)

**Comment by jonasrosland** [comment]
YouTube recording: https://youtu.be/RPGpqWGIzBs
